Automattic/wp-calypso
# Header Footer: move WordPress.com header and footer to package

## Proposed Changes

With the recent projects to unify our WordPress.com sites such as /support, /forums, and /learn we are lacking a unified approach to loading the constants like the header and footer. Both of these components have existed inside Calypso for the logged-out view of themes and plugins.

This PR creates a new package `packages/wpcom-template-parts` with to components inside, the header and footer. The goal from after this is to use them inside a block found in `apps/happy-blocks`. This PR depends on a separate PR to move the WordPressWatermark to components #73583

<img width="1906" alt="Markup 2023-02-20 at 17 44 41" src="https://user-images.githubusercontent.com/33258733/220162505-1e1c7eec-42b8-416b-b53d-55685266a937.png">

<img width="1897" alt="Markup 2023-02-20 at 17 45 01" src="https://user-images.githubusercontent.com/33258733/220162528-e1d025a0-b747-4d54-a02f-c9c145ac6036.png">

## Testing Instructions

This only adds a new package so there is no place to see it working yet. Please read through the code and look for errors. There will be a following PR to implement this.

## Pre-merge Checklist
